tests/Queue/QueueSyncTransactionalJobTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Illuminate\Tests\Queue;
+
+use Illuminate\Queue\SyncQueue;
+use PHPUnit\Framework\TestCase;
+use Illuminate\Queue\Jobs\SyncJob;
+use Illuminate\Container\Container;
+use Illuminate\Database\Connection;
+use Illuminate\Database\Capsule\Manager;
+use Illuminate\Contracts\Queue\Transactional;
+
+class QueueTransactionalJobTest extends TestCase
+{
+    /**
+     * @param $jobClassname
+     * @param $isTransactional
+     * @throws \ReflectionException
+     *
+     * @dataProvider jobsDataProvider
+     */
+    public function testRunFireInDatabaseTransaction($jobClassname, $isTransactional, $connectionName)
+    {
+        $connection = $this->createMock(Connection::class);
+        $connection->expects($isTransactional ? $this->once() : $this->never())
+            ->method('transaction')
+            ->willReturnCallback(function ($handler) {
+                $handler();
+            });
+
+        $db = $this->createMock(Manager::class);
+        $db->expects($isTransactional ? $this->once() : $this->never())
+            ->method('getConnection')
+            ->with($connectionName)
+            ->willReturn($connection);
+
+        /** @var Container|\PHPUnit_Framework_MockObject_MockObject $container */
+        $container = $this->createMock(Container::class);
+        $container->expects($this->exactly($isTransactional ? 2 : 1))
+            ->method('make')
+            ->willReturnCallback(function ($abstract) use ($db) {
+                switch ($abstract) {
+                    case 'db':
+                        return $db;
+                    default:
+                        return new $abstract;
+                }
+            });
+
+        $payloadFactory = new \ReflectionMethod(SyncQueue::class, 'createPayload');
+        $payloadFactory->setAccessible(true);
+
+        $payload = $payloadFactory->invoke(new SyncQueue(), $jobClassname, null, '');
+
+        unset($_SERVER['0(*_*)0']);
```

Comment:
Can we do better then working with a global variable here?

Reply:
Yep, job can use any payload, given from container mock to detect works.
But this is a simple test like Taylor have used [here](https://github.com/laravel/framework/blob/5.7/tests/Queue/QueueSyncQueueTest.php#L21)

It also can use real DB connection, real transactions. It's only base ones.